From b1a3df05f350359ce9e94f46fcc539c46d8e7908 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Wed, 30 Apr 2025 00:06:26 +0800
Subject: [PATCH 3/5] Ignore executable's RUNPATH for LD_AUDIT libraries

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/dl-load.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/elf/dl-load.c b/elf/dl-load.c
index f905578a..0e6452a2 100644
--- a/elf/dl-load.c
+++ b/elf/dl-load.c
@@ -2009,6 +2009,7 @@ _dl_map_object (struct link_map *loader, const char *name,
 
 	  /* Also try DT_RUNPATH in the executable for LD_AUDIT dlopen
 	     call.  */
+	  /*
 	  if (__glibc_unlikely (mode & __RTLD_AUDIT)
 	      && fd == -1 && !did_main_map
 	      && main_map != NULL && main_map->l_type != lt_loaded)
@@ -2021,6 +2022,7 @@ _dl_map_object (struct link_map *loader, const char *name,
 				&realname, &fb, loader ?: main_map,
 				LA_SER_RUNPATH, &found_other_class);
 	    }
+	  */
 	}
 
       /* Try the LD_LIBRARY_PATH environment variable.  */
-- 
2.49.0

