From 11edd6d8af9e65d2d23db8ba1743a9f684ac25ae Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Sat, 24 May 2025 17:34:19 +0000
Subject: [PATCH 3/3] Always search glibc libraries in CREW_GLIBC_PREFIX

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/dl-load.c | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/elf/dl-load.c b/elf/dl-load.c
index f905578a..eb1d85c2 100644
--- a/elf/dl-load.c
+++ b/elf/dl-load.c
@@ -114,6 +114,29 @@ static const size_t system_dirs_len[] =
 };
 #define nsystem_dirs_len array_length (system_dirs_len)
 
+static const char *glibc_libraries[] = {
+  "libBrokenLocale.so.1",
+  "libanl.so.1",
+  "libc.so.6",
+  "libc_malloc_debug.so.0",
+  "libdl.so.2",
+  "libm.so.6",
+  "libmemusage.so",
+  "libmvec.so.1",
+  "libnsl.so.1",
+  "libnss_compat.so.2",
+  "libnss_db.so.2",
+  "libnss_dns.so.2",
+  "libnss_files.so.2",
+  "libnss_hesiod.so.2",
+  "libpcprofile.so",
+  "libpthread.so.0",
+  "libresolv.so.2",
+  "librt.so.1",
+  "libthread_db.so.1",
+  "libutil.so.1"
+};
+
 static bool
 is_trusted_path_normalize (const char *path, size_t len)
 {
@@ -1548,6 +1571,20 @@ open_verify (const char *name, int fd,
   const char *errstring = NULL;
   int errval = 0;
 
+  if (name[0] != '/') {
+    for (int i = 0; i < sizeof(glibc_libraries) / sizeof(char *); i++) {
+      if (strcmp(name, glibc_libraries[i]) == 0) {
+        char *new_name = malloc(PATH_MAX);
+
+        strcpy(new_name, CREW_GLIBC_PREFIX "/");
+        strcat(new_name, name);
+
+        name = new_name;
+      }
+    }
+  }
+
+
 #ifdef SHARED
   /* Give the auditing libraries a chance.  */
   if (__glibc_unlikely (GLRO(dl_naudit) > 0))
@@ -1944,6 +1981,19 @@ _dl_map_object (struct link_map *loader, const char *name,
   /* Will be true if we found a DSO which is of the other ELF class.  */
   bool found_other_class = false;
 
+  if (name[0] != '/') {
+    for (int i = 0; i < sizeof(glibc_libraries) / sizeof(char *); i++) {
+      if (strcmp(name, glibc_libraries[i]) == 0) {
+        char *new_name = malloc(PATH_MAX);
+
+        strcpy(new_name, CREW_GLIBC_PREFIX "/");
+        strcat(new_name, name);
+
+        name = new_name;
+      }
+    }
+  }
+
 #ifdef SHARED
   /* Give the auditing libraries a chance to change the name before we
      try anything.  */
-- 
2.49.0

