From 1f86f02416907007ae98d7564fb049a6fb13e440 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Wed, 30 Apr 2025 00:13:59 +0800
Subject: [PATCH 3/3] Hardcode LD_LIBRARY_PATH with CREW_GLIBC_PREFIX included

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/rtld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/rtld.c b/elf/rtld.c
index 00bec153..db5fb5ab 100644
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -2647,7 +2647,7 @@ process_envvars_default (struct dl_main_state *state)
 	  /* The library search path.  */
 	  if (memcmp (envline, "LIBRARY_PATH", 12) == 0)
 	    {
-	      state->library_path = &envline[13];
+	      state->library_path = CREW_LD_LIBRARY_PATH;
 	      state->library_path_source = "LD_LIBRARY_PATH";
 	      break;
 	    }
-- 
2.49.0

