From 588f18a5f1e65d32412310760ea0daf6db74692e Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Sat, 24 May 2025 17:34:19 +0000
Subject: [PATCH 3/3] Always search glibc libraries in CREW_GLIBC_PREFIX

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/dl-load.c | 47 ++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 46 insertions(+), 1 deletion(-)

diff --git a/elf/dl-load.c b/elf/dl-load.c
index f905578a..cd5bcc9c 100644
--- a/elf/dl-load.c
+++ b/elf/dl-load.c
@@ -114,6 +114,37 @@ static const size_t system_dirs_len[] =
 };
 #define nsystem_dirs_len array_length (system_dirs_len)
 
+static const char *glibc_libraries[] = {
+  "libBrokenLocale.so.1",
+  "libanl.so.1",
+  "libc.so.6",
+  "libc_malloc_debug.so.0",
+  "libdl.so.2",
+  "libm.so.6",
+  "libmemusage.so",
+  "libmvec.so.1",
+  "libnsl.so.1",
+  "libnss_compat.so.2",
+  "libnss_db.so.2",
+  "libnss_dns.so.2",
+  "libnss_files.so.2",
+  "libnss_hesiod.so.2",
+  "libpcprofile.so",
+  "libpthread.so.0",
+  "libresolv.so.2",
+  "librt.so.1",
+  "libthread_db.so.1",
+  "libutil.so.1"
+};
+
+static bool is_glibc_library(const char *name) {
+  for (int i = 0; i < sizeof(glibc_libraries) / sizeof(char *); i++) {
+    if (strcmp(name, glibc_libraries[i]) == 0) return true;
+  }
+
+  return false;
+}
+
 static bool
 is_trusted_path_normalize (const char *path, size_t len)
 {
@@ -1961,7 +1992,21 @@ _dl_map_object (struct link_map *loader, const char *name,
     }
 #endif
 
-  if (strchr (name, '/') == NULL)
+  if (is_glibc_library(name))
+    {
+      if (__glibc_unlikely (GLRO(dl_debug_mask) & DL_DEBUG_LIBS))
+  _dl_debug_printf ("find library=%s [%lu]; searching in CREW_GLIBC_PREFIX (%s)\n", name, nsid, CREW_GLIBC_PREFIX);
+
+      realname = malloc(PATH_MAX);
+
+      strcpy(realname, CREW_GLIBC_PREFIX "/");
+      strcat(realname, name);
+
+      fd = open_verify(realname, -1, &fb,
+        loader ?: GL(dl_ns)[nsid]._ns_loaded, 0, mode,
+        &found_other_class, true);
+    }
+  else if (strchr (name, '/') == NULL)
     {
       /* Search for NAME in several places.  */
 
-- 
2.49.0

