From f757ee5e8673db382221626bb656776d73be89a2 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Thu, 15 May 2025 02:02:41 +0800
Subject: [PATCH 3/3] Always append CREW_GLIBC_PREFIX to LD_LIBRARY_PATH

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/rtld.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/elf/rtld.c b/elf/rtld.c
index 00bec153..ee75fd80 100644
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -2559,6 +2559,13 @@ process_envvars_default (struct dl_main_state *state)
   char **runp = _environ;
   char *envline;
   char *debug_output = NULL;
+  char *end_of_library_path;
+
+  state->library_path        = malloc(PATH_MAX * 32);
+  state->library_path_source = "LD_LIBRARY_PATH";
+  end_of_library_path        = (char *) state->library_path + strlen(CREW_GLIBC_PREFIX);
+
+  strcpy((char *) state->library_path, CREW_GLIBC_PREFIX);
 
   while ((envline = _dl_next_ld_env_entry (&runp)) != NULL)
     {
@@ -2647,8 +2654,10 @@ process_envvars_default (struct dl_main_state *state)
 	  /* The library search path.  */
 	  if (memcmp (envline, "LIBRARY_PATH", 12) == 0)
 	    {
-	      state->library_path = &envline[13];
-	      state->library_path_source = "LD_LIBRARY_PATH";
+	      if (strlen(&envline[13]) + (end_of_library_path - state->library_path) < PATH_MAX * 32) {
+	        strcpy(end_of_library_path++, ":");
+	        strcpy(end_of_library_path, &envline[13]);
+	      }
 	      break;
 	    }
 
-- 
2.49.0

