From c13a921e90e362531f986e6707dd30023295bc57 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Tue, 29 Apr 2025 23:10:01 +0800
Subject: [PATCH 4/4] Hardcode LD_LIBRARY_PATH with CREW_GLIBC_PREFIX included

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 elf/rtld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/rtld.c b/elf/rtld.c
index 2d41ed1f..5adfa1a5 100644
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -2650,7 +2650,7 @@ process_envvars_default (struct dl_main_state *state)
 	  /* The library search path.  */
 	  if (memcmp (envline, "LIBRARY_PATH", 12) == 0)
 	    {
-	      state->library_path = &envline[13];
+	      state->library_path = CREW_LD_LIBRARY_PATH;
 	      state->library_path_source = "LD_LIBRARY_PATH";
 	      break;
 	    }
-- 
2.49.0

